src/azure/basic-azure-cli-facade.ts

```diff
@@ -90,6 +95,28 @@ export class BasicAzureCliFacade implements AzureCliFacade {
     return JSON.parse(result.stdout) as ConsumptionInfo[];
   }
 
+  async getRoleAssignments(
+    subscription: Subscription,
+  ): Promise<RoleAssignment[]> {
+    const queryFields = [
+      "principalId",
+      "principalName",
+      "principalType",
+      "roleDefinitionId",
+      "roleDefinitionName",
+    ];
+    const jsonQuery = queryFields.map((x) => x + ":" + x).join(",");
+    //  --query '[].{${jsonQuery}}' -> could be added but stdout gets weird?
```

Comment:
@tfelix i remember that this didn't work and my output resulted in the actual jsonQuery? It was super weird, maybe you know why this happened.

Reply:
not sure but az cli has very tricky ways of use arguments. I also struggled half a day to figure out how to pipe in JSON input. Might have something to do with its quotes around it? It probably would work out if you do some tests and variations on how to pass that argument. But as long as it works in the end.